packaging/common/substitute.pl

```diff
@@ -56,21 +53,163 @@ sub fix_pkg_rev {
+    chomp $RELTYPES;
+
+    $RELTYPES ||= qx{exec 2>/dev/null; . /etc/os-release; . /etc/lsb-release; echo -n "\$ID \$DISTRIB_ID"};
+    chomp $RELTYPES;
+
+    $RELTYPES ||= qx{exec 2>/dev/null; command ls -1 /etc/\{debian*,*release\}};
+    chomp $RELTYPES;
+
+    # change to all lowercase
+    $RELTYPES = lc($RELTYPES);
+    $RELTYPES =~ s/\n//sg;
+
+    my $RELVER = qx{exec 2>/dev/null; lsb_release --release};
+    chomp $RELVER;
+
+    $RELVER ||= qx{exec 2>/dev/null; cat /etc/debian_version};
+    chomp $RELVER;
+
+    # allow all digits for solaris or centos or rhel 
+    $RELVER ||= qx{exec 2>/dev/null; cat /etc/release /etc/centos-release /etc/redhat-release };
+    chomp $RELVER;
+
+    # get an obvious ident and use first-digit-range only
+    $RELVER ||= qx{exec 2>/dev/null; . /etc/os-release; . /etc/lsb-release; echo "\$VERSION_ID" };
+    chomp $RELVER;
+    
+    $RELVER =~ s/\n//sg;
+    $RELVER =~ s/^\D*//; 
+    $RELVER =~ s/\D*$//; 
+    chomp $RELVER;
+
+    # remove leading non-digits and toss all after first non-digit...
+    my $RELVER1 = $RELVER;
+    my $RELVER2 = $RELVER;
+
+    $RELVER1 =~ s/\D.*$//;
+
+    # remove leading non-digits and toss all after second non-digit...
+    $RELVER2 =~ s/\D(\d+)/$1/; 
+    $RELVER2 =~ s/\D.*$//;
+
+    $arch="amd64" 
+       if ( $RELTYPES =~ m/debian|ubuntu/i && $arch eq "x86_64" );
+    $arch="i386" 
+       if ( $RELTYPES =~ m/debian|ubuntu/i && $arch eq "i686" );
+
+    return [ ".$RELVER.pkg", "SunOS", $RELVER ]
+       if ( $RELTYPES eq "/etc/release" );
+
+    return [ ".fc${RELVER1}.${arch}.rpm", "Fedora", $RELVER1 ]
+       if ( $RELTYPES =~ m/fedora/ );
+
+    return [ ".rhel${RELVER1}.${arch}.rpm", "Centos", $RELVER1 ]
+       if ( $RELTYPES =~ m/centos/ );
+
+    # mention redhat only last
+    return [ ".rhel${RELVER1}.${arch}.rpm", "RHEL", $RELVER1 ]
+       if ( $RELTYPES =~ m/redhat/ );
+
+    return [ ".sles${RELVER2}.${arch}.rpm", "SLES", $RELVER2 ]
+       if ( $RELTYPES =~ m/\bsuse\b/ && $RELTYPES =~ m/\benterprise\b/ );
+
+    return [ ".suse${RELVER2}.${arch}.rpm", "SuSE", $RELVER2 ]
+       if ( $RELTYPES =~ m/opensuse/ );
+
+    return [ "Ubuntu${RELVER2}_${arch}.deb", "Ubuntu", $RELVER2 ]
+       if ( $RELTYPES =~ m/ubuntu/ );
+
+    # mention debian after ubuntu
+    return [ "Debian${RELVER1}_${arch}.deb", "Debian", $RELVER1 ]
+       if ( $RELTYPES =~ m/debian/ );
+
+    return [ ];
+}
+
+sub get_platform_info {
+    return [ $ENV{"PLATFORM_PKG"}, $ENV{"PLATFORM_DIST"}, $ENV{"PLATFORM_DISTVER"} ]
+       if ( $ENV{"PLATFORM_PKG"} && $ENV{"PLATFORM_DIST"} && $ENV{"PLATFORM_DISTVER"} );
+
+    my $arrayp = find_platform_info();
+    my ($suffix,$dist,$distver) = @{$arrayp};
+
+    $ENV{"PLATFORM_PKG"} = $suffix if ( ! $ENV{"PLATFORM_PKG"} );
+    $ENV{"PLATFORM_DIST"} = $dist if ( ! $ENV{"PLATFORM_DIST"} );
+    $ENV{"PLATFORM_DISTVER"} = $distver if ( ! $ENV{"PLATFORM_DISTVER"} );
+    return $arrayp;
+}
+
+use constant PLATFORM_GLIB2_VERSIONS => qw( 
```

Comment:
this is needed because GLIB2 compiled versions (for Amanda) cannot be reordered to get any earlier version of Glib2.

all these refer to a specific Linux distro/rev that starts with the *earliest* Glib rev, which must be installed to be the most compatible.